### PR TITLE
Refactor(eos_designs): router_ospf python_module as per eos_cli_config_gen

### PR DESCRIPTION
**Pull request by Vibhu-gslab**
## Change Summary

Refactor router_ospf python_module in core_interfaces, l3_edge, network_services and underlay modules as per eos_cli_config_gen

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_ospf.py [hunk 1]
```diff
@@ -69,16 +69,17 @@ def router_ospf(self) -> dict | None:
                     if (route_map := get(vrf, "ospf.redistribute_bgp.route_map")) is not None:
                         process["redistribute"]["bgp"]["route_map"] = route_map
 
+                process["id"] = process_id
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_ospf.py]: just set this as part of the definition of `process` in line 56

File: ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_ospf.py [hunk 2]
```diff
@@ -69,16 +69,17 @@ def router_ospf(self) -> dict | None:
...
 
-                ospf_processes[process_id] = process
+                ospf_processes.append(process)
 
         # If we have static_routes in default VRF and not EPVN, and underlay is OSPF
         # Then add redistribute static to the underlay OSPF process.
         if self._vrf_default_ipv4_static_routes["redistribute_in_underlay"] and self._underlay_routing_protocol in ["ospf", "ospf-ldp"]:
-            ospf_processes[self._underlay_ospf_process_id] = {"redistribute": {"static": {}}}
-
+            ospf_processes.append(dict({"redistribute": {"static": {}}}, id=int(self._underlay_ospf_process_id)))
```
Comment (ClausHolbechArista) [inline ansible_collections/arista/avd/roles/eos_designs/python_modules/network_services/router_ospf.py]:
```suggestion
            ospf_processes.append({"id": self._underlay_ospf_process_id, "redistribute": {"static": {}}})
```

**Review by ClausHolbechArista — CHANGES_REQUESTED** [review]

**Review by emilarista — APPROVED** [review]
LGTM!

**Review by tgodaA — APPROVED** [review]
LGTM

**Review by gmuloc — APPROVED** [review]
LGTM

**Review by ClausHolbechArista — APPROVED** [review]

**Pull request merged**